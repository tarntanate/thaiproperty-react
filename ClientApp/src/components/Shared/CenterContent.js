import React from 'react';
import { Panel } from 'react-bootstrap';

const MARGIN_TOP = '20px';

export const CenterContent = props => {
  if (props.usePanel) {
    return (
      <Panel onClick={props.handlePanelClick} style={{ marginTop: MARGIN_TOP }}>
        <Panel.Body
          style={{
            width: '100%',
            textAlign: 'center',
            verticalAlign: 'middle',
          }}
        >
          {props.children}
        </Panel.Body>
      </Panel>
    );
  } else {
    return (
      <div
        style={{
          width: '100%',
          textAlign: 'center',
          verticalAlign: 'middle',
          marginTop: MARGIN_TOP,
        }}
      >
        {props.children}
      </div>
    );
  }
};
