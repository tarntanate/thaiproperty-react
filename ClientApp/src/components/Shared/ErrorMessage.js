import React from 'react';
import Alert from 'antd/lib/alert';

// a re-usable presentation component for showing error message.
export const ErrorMessage = ({text}) => (
  <div style={{ margin: 20 }}>
    <Alert message="มีข้อผิดพลาด" description={text} type="error" showIcon />
  </div>
);