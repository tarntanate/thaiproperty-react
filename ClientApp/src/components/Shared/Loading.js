import React from 'react';
import Spin from 'antd/lib/spin';
import Icon from 'antd/lib/icon';

const customIcon = <Icon type="loading" style={{ fontSize: 24 }} spin />;
const defaultClass = 'loader m-3 text-center';

export const Loading = ({text = 'Loading more data...', className, ...props}) => (
    <div className={[defaultClass, className].join(' ')} >
        <Spin indicator={customIcon} style={{ marginRight: 10 }} />{text}
    </div>
);