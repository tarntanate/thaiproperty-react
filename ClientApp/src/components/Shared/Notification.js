import notification from 'antd/lib/notification';
import {
    SHOW_MESSAGE_DURATION
} from '../../config';

export const openNotification = ({
    message,
    description,
    type = 'open',
    duration = SHOW_MESSAGE_DURATION
}) => {
    notification.config({
        placement: 'bottomRight',

        duration,
    });

    notification[type]({
        message: message,
        description: description,
    });
};