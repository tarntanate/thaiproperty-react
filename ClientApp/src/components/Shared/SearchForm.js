import React, { Component } from 'react';
import { connect } from 'react-redux';
// Total antd module size is 324K gzipped, so we must selectively import
import Card from 'antd/lib/card/';
import Form from 'antd/lib/form/';
import Button from 'antd/lib/button/';
import Input from 'antd/lib/input/';
import Icon from 'antd/lib/icon/';

const FormItem = Form.Item;

class SearchForm extends Component {
  state = {
    search: '',
  };

  handleFormSubmit = e => {
    e.preventDefault();
    this.props.handleFormSubmit();
  };

  render() {
    return (
      <Card bordered={false}>
        <Form layout="inline" onSubmit={this.handleFormSubmit}>
          <FormItem wrapperCol={{ span: 24 }}>
            <Input
              type="text"
              size="large"
              placeholder={this.props.placeholderText}
              value={this.state.search}
              onChange={e => {
                this.setState({ search: e.target.value });
                if (this.props.handleTextChanged) {
                  this.props.handleTextChanged();
                }
              }}
            />
            <Button type="primary" onClick={e => this.handleFormSubmit(e)}>
              <Icon type="search" />
              {this.props.buttonText}
            </Button>
          </FormItem>
        </Form>
      </Card>
    );
  }
}

SearchForm.defaultProps = {
  placeholderText: 'ใส่คำค้นหา...',
  buttonText: 'ค้นหา',
};

function mapStateToProps(state) {
  return { ...state };
}

export default connect(mapStateToProps)(SearchForm);
