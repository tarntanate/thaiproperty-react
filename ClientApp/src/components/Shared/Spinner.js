import React from 'react';
import Spin from 'antd/lib/spin';

export const Spinner = ({size = 'large', style}) => (
  <Spin size={size} style={style} />
)