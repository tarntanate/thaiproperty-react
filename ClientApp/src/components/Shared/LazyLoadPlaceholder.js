import React from 'react';
import { Spinner } from '../Shared/Spinner';

export const PlaceHolder = (props) => (
    <div style={{margin:20}}>
        <Spinner size="small" style={{marginRight:10}} {...props} /> 
        <span className="text-muted">Loading image...</span>
    </div>
)