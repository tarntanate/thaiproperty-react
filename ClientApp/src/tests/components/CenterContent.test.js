import React from 'react';
import renderer from 'react-test-renderer';
import { render, shallow } from 'enzyme';
import { CenterContent } from '../../components/Shared/CenterContent';

describe('CenterContent Components', () => {
  it('Should renders match snapshot', () => {
    const component = renderer
      .create(<CenterContent usePanel><h2>Inside Content</h2></CenterContent>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Should renders correctly without \'panel\' className when without usePanel', () => {
    let component = render(<CenterContent><h2>Inside Content</h2></CenterContent>);
    expect(component.hasClass('panel')).toEqual(false);
  });
  
  it('Should renders correctly with bootstrap className \'panel\' with usePanel', () => {
    let component = render(<CenterContent usePanel><h2>Inside Content</h2></CenterContent>);
    expect(component.hasClass('panel')).toEqual(true);
  });

  it('Shallow rendering', () => {
    let componentFromShallow = shallow(<CenterContent usePanel><h2>Inside Content</h2></CenterContent>);
    console.log('componentFromShallow');
    console.log(componentFromShallow);
  });
});