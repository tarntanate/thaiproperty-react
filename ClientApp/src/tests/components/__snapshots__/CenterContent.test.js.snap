// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterContent Components Should renders match snapshot 1`] = `
<div
  className="panel panel-default"
  onClick={undefined}
  style={
    Object {
      "marginTop": "20px",
    }
  }
>
  <div
    className="panel-body"
    style={
      Object {
        "textAlign": "center",
        "verticalAlign": "middle",
        "width": "100%",
      }
    }
  >
    <h2>
      Inside Content
    </h2>
  </div>
</div>
`;
